src/Slider.jsx

```diff
@@ -159,123 +182,118 @@ class Rheostat extends React.Component {
     this.slideTo = this.slideTo.bind(this);
     this.updateNewValues = this.updateNewValues.bind(this);
     this.setRef = this.setRef.bind(this);
-    this.invalidatePitStyleCache = this.invalidatePitStyleCache.bind(this);
+    this.setHandleNode = this.setHandleNode.bind(this);
+    this.setHandleContainerNode = this.setHandleContainerNode.bind(this);
+    this.positionPercent = this.positionPercent.bind(this);
+  }
 
-    this.pitStyleCache = {};
+  componentDidMount() {
+    // Note: This occurs in a timeout because styles need to be applied first
+    this.handleDimensionsTimeout = setTimeout(() => {
+      this.handleDimensionsTimeout = null;
+      this.setState({ handleDimensions: this.getHandleDimensions() });
+    }, 0);
   }
 
   componentWillReceiveProps(nextProps) {
     const {
-      className,
       disabled,
       min,
       max,
-      orientation,
-      pitPoints,
-      algorithm,
     } = this.props;
+
     const {
       values,
       slidingIndex,
     } = this.state;
 
-    const minMaxChanged = (nextProps.min !== min || nextProps.max !== max);
+    const minMaxChanged = (
+      nextProps.min !== min || nextProps.max !== max
+    );
 
     const valuesChanged = (
       values.length !== nextProps.values.length ||
       values.some((value, idx) => nextProps.values[idx] !== value)
     );
 
-    const orientationChanged = (
-      nextProps.className !== className ||
-      nextProps.orientation !== orientation
-    );
-
-    const algorithmChanged = nextProps.algorithm !== algorithm;
-
-    const pitPointsChanged = nextProps.pitPoints !== pitPoints;
-
     const willBeDisabled = nextProps.disabled && !disabled;
 
-    if (orientationChanged) {
-      this.setState({
-        className: getClassName(nextProps),
-      });
-    }
-
     if (minMaxChanged || valuesChanged) this.updateNewValues(nextProps);
 
-    if (minMaxChanged || pitPointsChanged || orientationChanged || algorithmChanged) {
-      this.invalidatePitStyleCache();
-    }
-
     if (willBeDisabled && slidingIndex !== null) {
       this.endSlide();
     }
   }
 
+  componentWillUnmount() {
+    if (this.handleDimensionsTimeout) {
+      clearTimeout(this.handleDimensionsTimeout);
+    }
+  }
+
   getPublicState() {
-    const { min, max } = this.props;
     const { values } = this.state;
+    const {
+      min,
+      max,
+    } = this.props;
 
-    return { max, min, values };
+    return {
+      max,
+      min,
+      values,
+    };
   }
 
-  // istanbul ignore next
   getSliderBoundingBox() {
-    const { rheostat } = this;
-    const node = rheostat.getDOMNode ? rheostat.getDOMNode() : rheostat;
-    const rect = node.getBoundingClientRect();
-
+    const rect = this.handleContainerNode.getBoundingClientRect();
     return {
-      height: rect.height || node.clientHeight,
+      height: rect.height || this.handleContainerNode.clientHeight,
```

Comment:
@ljharb Did we figure out whether or not we need to check whether or not refs are null?

Reply:
they will only be null if:
 - you pass a ref callback that changes identity across renders
 - the component is not yet mounted / unmounted
 - the component the ref points to is conditionally rendered (which is a subset of above)